pkg/lint/rules/template.go

```diff
@@ -94,7 +94,7 @@ func Templates(linter *support.Linter) {
 		// NOTE, disabled for now, Refs https://github.com/kubernetes/helm/issues/1037
 		// linter.RunLinterRule(support.WarningSev, path, validateQuotes(string(preExecutedTemplate)))
 
-		renderedContent := renderedContentMap[fileName]
+		renderedContent := renderedContentMap[filepath.Join(chart.GetMetadata().Name, fileName)]
```

Comment:
not sure if this is the best way to add the chart directory to the path, but since we enforce the chart name being the same as the directory name, it should be ok.